temporalcli/internal/cmd/gen-docs/main.go

```diff
@@ -32,6 +32,11 @@ func run() error {
 		return fmt.Errorf("failed parsing markdown: %w", err)
 	}
 
+	cmds, err = commandsgen.EnrichCommands(cmds)
```

Comment:
Can the logic for this just become part of docs gen or parsing or whatever? Why a separate call?

Reply:
It could for sure.

Considerations, IMO:
- `parse.go` is more about reading from`commands.yml` content and doing field validation
- `enrich.go` is about adding fields that aren't read from `commands.yml` to make it easier for `docs.go`
- also this https://github.com/temporalio/tcld/pull/398
  - port docs gen to `tcld`
  - skipped `ParseCommands()` since [we do `ConvertCommands()` -> `EnrichCommands()`](https://github.com/temporalio/tcld/pull/398/files#diff-de51fa3d61c20d869a43ce7264592eb7bef72af035915ba01b80277f00f1e043R31)
  - we do [emit a tcld.yml now](https://github.com/temporalio/tcld/pull/398/files#diff-4e3ba5b9cad41fc0a5def1b128823a7ed408f7fd3a0e392e25cbf2e190d0715c)
   - maybe do want to add front matter to `tcld.yml` and round trip in a way compatible with `ConvertCommands` so maybe we will need a `ParseCommands` over there, for example
      - `ParseCommands()` -> `ConvertCommands()` -> save `tcld.yml`-> `EnrichCommands()`
        - convert would overlay from cli.Command content onto `Commands`
        - round trip writes back to `tcld.yml` in a way the doesn't wipe out docs front matter
        - don't want enriched fields being written to `tcld.yml`
  - so wanted something more consistent across both, in the event we could extract a common package.

Reply:
> enrich.go is about adding fields that aren't read from commands.yml to make it easier for docs.go

If code is only used for docs.go then just put it as part of docs.go. However, at parse time, it's totally fine to populate things that are not necessarily used by everyone. This is a codegen step, feel free to add extra things to parse.

> port docs gen to tcld

`tcld` is beta software that may (will likely) be completely replaced to fit in this CLI model, so I wouldn't spend too much effort on it nor would I affect anything in this repo with anything done over there

Reply:
> However, at parse time, it's totally fine to populate things that are not necessarily used by everyone. Feel free to add extra things to parse and/or extra fields to the command struct that only docs side will use.

Sounds good. Moved the enrichments to `parse.go`.

Reply:
Made a couple of comments. IMO we should only add fields to the parsed field list that are not otherwise obtainable and are needed. Otherwise, obtain or pass around at runtime.

Reply:
Fixed.